packages/components/src/components/teaser/teaser.test.ts

```diff
@@ -184,4 +186,38 @@ describe('<sd-teaser>', () => {
     expect(el.variant).to.equal('neutral-100');
     expect(el.inset).to.equal(false); // The inset should be false after the variant change
   });
+
+  describe('reversed layout', () => {
+    it('works properly in horizontal mode', async () => {
+      const el = await fixture<SdTeaser>(html`
+        <sd-teaser reversed-layout>
+          <div slot="media">Media</div>
+          <div slot="headline">Headline</div>
+          Content
+        </sd-teaser>
+      `);
+
+      el._orientation = 'horizontal';
+      await el.updateComplete;
+
+      const media = el.shadowRoot!.querySelector('#media')!;
+      expect(media.classList.contains('order-2')).to.be.true;
+    });
+
+    it('is not applied if the orientation is vertical', async () => {
+      const el = await fixture<SdTeaser>(html`
+        <sd-teaser reversed-layout>
+          <div slot="media">Media</div>
+          <div slot="headline">Headline</div>
+          Content
+        </sd-teaser>
+      `);
+
+      el._orientation = 'vertical';
+      await el.updateComplete;
+
+      const media = el.shadowRoot!.querySelector('#media')!;
+      expect(media.classList.contains('order-2')).to.be.false;
```

Comment:
IMO these tests can be removed. We have visual tests to check, if everything is fine. It's only important to check here, if the default is correct, and that's what you already did.

Furthermore you should only set and unset`reversed-layout` instead of hijacking `el._orientation` for that.